cpp/src/io/parquet/reader_impl.hpp

```diff
@@ -354,6 +321,25 @@ class reader::impl {
   }
 
  private:
+  /**
+   * @brief Check if the user has specified custom row bounds
+   *
+   * @param read_mode Value indicating if the data sources are read all at once or chunk by chunk
+   * @return True if the user has specified custom row bounds
+   */
+  [[nodiscard]] bool uses_custom_row_bounds(read_mode mode) const
+  {
+    // TODO: The following should the same for both `read_modes` and not hardcoded to true for
+    // `read_mode::CHUNKED_READ`. We have done this for now due to following: We are calling
+    // `read_chunk_internal()` in a few places where we're only passing `uses_custom_row_bounds`.
+    // But if chunked_read_size is > 0 we need to force that to be true. If we don't, various
+    // important things don't happen for all the remaining chunks. Basically if this code doesn't
+    // execute, it will be bad.
```

Comment:
Please refine the comments :smile: .

Reply:
I was on the fence on this one as well :D

Reply:
All done!